### PR TITLE
docs: :memo: include comparison of languages in Why Python post

### LINKED ISSUE #7: design/decisions/why-python: Does not include a presentation on the alternatives and why these were ruled out
The other decision pages presents benefits and drawbacks of the alternatives one by one. Furthermore, it is presented why these alternatives were considered. Shouldn't that be added here as well? :-)

It is only stated which alternatives were considered. Not *why* these particular ones:
https://github.com/seedcase-project/seedcase-project/blob/eb78a8204cb56f1471dffbb7bab8f0b6f1ea6d43/design/decisions/why-python/index.qmd#L23-L26

It does include this shorter argumentation for the drawbacks of the alternatives:
https://github.com/seedcase-project/seedcase-project/blob/eb78a8204cb56f1471dffbb7bab8f0b6f1ea6d43/design/decisions/why-python/index.qmd#L48-L52

### PR DESCRIPTION
**Pull request by martonvago**
## Description

- This PR updates the Why Python post, adding pros and cons for the languages considered.
- These changes are done to standardise the structure of decision posts and to add more context.

## Related Issues

Closes #7 

See also Issue #5 

## Reviewer Focus

This PR only needs a quick review.

## Checklist

For general documentation:

- [ ] Spell-check
    - [ ] US
    - [x] UK
- [x] Did the page(s) preview correctly on your machine without breaking
- [x] New category words (keywords) (if any) added to the code snippet file

### CODE REVIEW

File: why-python.qmd [hunk 1]
```diff
@@ -2,9 +2,8 @@
 title: "Why Python"
 description: |
   Python is one of the most common and widely used programming languages.
-  It is used across multiple domains and industries, which means more people 
-  would be familiar with using it.
-date: "2023-03-22"
+  As it is used across multiple domains and industries, it is likely to be familiar to a wider audience.
+date: "2024-05-10"
```
Comment (martonvago) [inline why-python.qmd:6]: I was supposed to update this, right?

File: why-python.qmd [hunk 2]
```diff
@@ -112,10 +139,17 @@ What decision was made, use the form "We decided on CHOICE because of
 REASONS."
 :::
 
+We have decided to use Python as the main development language for Seedcase because it is a powerful, flexible and easy-to-use platform for building a data management web application. Drawing on the skills of the core Seedcase team, Python offers the best balance of capabilities and ease of development out of the options considered. It is, moreover, a language likely to be familiar to technologically-oriented Seedcase users and prospective contributors.
+
+While Java and C++ offer more scope for context-specific performance optimisation, we are unlikely to need this level of control for Seedcase, because the number of concurrent user requests is expected to be low, resource-intensive backend tasks are expected to run without much competition, and we don't expect to do heavy-duty, real-time data analysis or image processing. Therefore, for our use case, the advantages of these languages don't justify their steeper learning curve and added complexity.
```
Comment (martonvago) [inline why-python.qmd:144]: Here I'm thinking that Seedcase won't serve thousands of user requests at once, won't run very many high-workload scheduled jobs in the background and won't do super complex calculations on data. But that's just my understanding so far, so let me know if that's not the case!

File: why-python.qmd (hunk 1 above)
Comment (K-Beicher) [reply why-python.qmd:6]: Yes.

File: why-python.qmd (hunk 2 above)
Comment (K-Beicher) [reply why-python.qmd:144]: Honestly sounds about right, but maybe @lwjohnst86 would like to comment as well.

**Review by K-Beicher — DISMISSED** [review]
Exactly what I was hoping you would produce in terms of format and layout. I'm not an expert on these things, so I'll leave it to the next reviewer to critique any points made, but I think they look spot on.

File: why-python.qmd (hunk 1 above)
Comment (lwjohnst86) [inline why-python.qmd:6]:
```suggestion
date: "2023-03-22"
date-modified: "2024-05-10"
```

File: why-python.qmd [hunk 3]
```diff
@@ -51,12 +50,24 @@ drawbacks for each option.
 ::: {.column}
 #### Benefits
 
-- Item 1
+- Well-suited for real-time and performance-critical applications.
+- Allows fine-grained control over memory and hardware resources, enabling low-level, use-case-specific optimisations.
+- Compilation to native machine code for the target platform prior to execution guarantees better runtime performance than Python and, potentially and by a smaller margin, Java.
```
Comment (lwjohnst86) [inline why-python.qmd:55]:
```suggestion
- Compilation to native machine code for the target platform prior to execution guarantees better runtime performance than Python and, potentially by a smaller margin, Java.
```

File: why-python.qmd [hunk 4]
```diff
@@ -81,12 +100,18 @@ drawbacks for each option.
 ::: {.column}
 #### Benefits
 
-- Item 1
+- Code is executed by a Python interpreter, making Python programs platform independent (provided the host has an interpreter installed).
+- Concise, easy-to-read, beginner-friendly syntax.
+- Easy to iterate and prototype rapidly.
+- Large ecosystem of mature libraries and frameworks for web development (e.g. [Django](https://www.djangoproject.com), [Flask](https://flask.palletsprojects.com/en/2.3.x/)), database management (e.g. [SQLAlchemy](https://www.sqlalchemy.org), Django's ORM), and data analysis.
```
Comment (lwjohnst86) [inline why-python.qmd:106]:
```suggestion
- Large ecosystem of mature libraries and frameworks for web development (e.g. [Django](https://www.djangoproject.com), [Flask](https://flask.palletsprojects.com/en/2.3.x/)), database management (e.g. [SQLAlchemy](https://www.sqlalchemy.org), Django's Object Relational Mapper), and data analysis.
```

File: why-python.qmd [hunk 5]
```diff
@@ -96,12 +121,16 @@ drawbacks for each option.
 ::: {.column}
 #### Benefits
 
-- Item 1
+- Designed for statistical computing with excellent data visualisation capabilities.
+- Wide array of libraries for data analysis.
```
Comment (lwjohnst86) [inline why-python.qmd:125]:
```suggestion
- Wide array of packages for data analysis.
```

File: why-python.qmd (hunk 2 above)
Comment (lwjohnst86) [reply why-python.qmd:146]: @martonvago you got it!!

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Very nice! 😁 Some small edits

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**